server/backends/migration_cache/migration_cache.go

```diff
@@ -0,0 +1,184 @@
+package migration_cache
+
+import (
+	"context"
+	"io"
+	"reflect"
+
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/backends/pebble_cache"
+	"github.com/buildbuddy-io/buildbuddy/server/backends/disk_cache"
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
+	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/util/flagutil"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+
+	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+	cache_config "github.com/buildbuddy-io/buildbuddy/server/cache/config"
+)
+
+var (
+	cacheMigrationConfig = flagutil.New("cache.migration", MigrationConfig{}, "Config to specify the details of a cache migration")
+)
+
+type MigrationCache struct {
+	Src  interfaces.Cache
+	Dest interfaces.Cache
+}
+
+func Register(env environment.Env) error {
+	if cacheMigrationConfig.Src == nil || cacheMigrationConfig.Dest == nil {
+		return nil
+	}
+	log.Infof("Registering Migration Cache")
+
+	srcCache, err := getCacheFromConfig(env, *cacheMigrationConfig.Src)
+	if err != nil {
+		return err
+	}
+	destCache, err := getCacheFromConfig(env, *cacheMigrationConfig.Dest)
+	if err != nil {
+		return err
+	}
+	mc := &MigrationCache{
+		Src:  srcCache,
+		Dest: destCache,
+	}
+
+	if env.GetCache() != nil {
+		log.Warningf("Overriding configured cache with migration_cache. If running a migration, all cache configs" +
+			" should be nested under the cache.migration block.")
+	}
+	env.SetCache(mc)
+
+	return nil
+}
+
+func validateCacheConfig(config CacheConfig) error {
+	numConfigs := 0
+	v := reflect.ValueOf(config)
+	for i := 0; i < v.NumField(); i++ {
+		if !v.Field(i).IsNil() {
+			numConfigs++
+		}
+	}
```

Comment:
Kinda gnarly to use reflection here! Can you just write this code out?

Reply:
Yeah thought that would make it more generalizable, but agree that it's probably just over-engineering